### PR TITLE
Feature/rdss 1784

### PR DESCRIPTION
**Pull request by camilla-synbyote**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -46,7 +46,7 @@ pytest-cov==2.5.1
 python-dateutil==2.6.1
 pytz==2018.3
 PyYAML==3.12
--e git://github.com/JiscRDSS/rdss-shared-libraries.git@b07c328bcb8a812be5ba76efeb3ac7439411e15e#egg=rdsslib
+-e git://github.com/JiscRDSS/rdss-shared-libraries.git@5684dd8fa8b06c8aeaafea9393c4a584325ae0ba#egg=rdsslib
```
Comment (michalc) [inline requirements.txt]: Just to check this is the correct commit? I'm not completely sure, but I don't think it has the changes from 3e24170d26749e0c645e551e4b64a1058002118b ?

File: startup.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (michalc) [inline startup.sh]: Just to confirm, this is definitely more preferred than running a cron on Jenkins itself? Just that we do have some crons on Jenkins, so just making sure that consistency in that respect has been explicitly rejected.

File: crontab [hunk 3]
```diff
@@ -1 +1 @@
-30 * * * * root /usr/local/bin/python3 /app/pure_adaptor/pure_adaptor.py > /proc/1/fd/1 2>/proc/1/fd/2
+*/20 * * * * root /usr/local/bin/python3 /app/pure_adaptor/pure_adaptor.py > /proc/1/fd/1 2>/proc/1/fd/2
```
Comment (michalc) [inline crontab]: Is this file still needed? There is some duplication between this and code in startup.sh?

File: pure_adaptor/pure/base/remapper.py [hunk 4]
```diff
@@ -14,7 +15,7 @@
 
 
 HEI_ADDRESS = {
-    799: 'University of St.Andrews, KY16 9AJ, Fife'
+    '799': 'University of St.Andrews, KY16 9AJ, Fife'
```
Comment (michalc) [inline pure_adaptor/pure/base/remapper.py]: Feel free to push back (especially if what I'm suggesting doesn't make sense!), but now that the JiscId is being passed in via an environment variable, maybe the address should be too? So when releasing the adapter for another institution, it wouldn't need _both_ a code change and a terraform change, we could probably just do it with a terraform change?

File: pure_adaptor/pure/v59/research_object_mapping.txt [hunk 5]
```diff
@@ -33,14 +33,25 @@
...
-                licenceIdentifier : documentLicense[0].uri
+                licenceName: documentLicense[0].value || 'not present',
+                licenceIdentifier : documentLicense[0].uri || 'not present'
                 }],
     access: [{
                 accessType: `1`,
                 accessStatement: 'not present'
                 }]
-  },
+  } || [{
```
Comment (michalc) [inline pure_adaptor/pure/v59/research_object_mapping.txt]: It looks like the right hand side of the `||` would never be used here? However, I'm not exactly sure what `||` does in this context

File: startup.sh (hunk 2 above)
Comment (camilla-synbyote) [reply startup.sh]: I'm probably not following something in this train of thought (also not an expert in Jenkins), but this script creates the cron that will run the Pure adaptor Python code in the Docker container on the ECS cluster.

File: crontab (hunk 3 above)
Comment (camilla-synbyote) [reply crontab]: You're right, it's not - sloppy commit from me!

File: pure_adaptor/pure/v59/research_object_mapping.txt (hunk 5 above)
Comment (camilla-synbyote) [reply pure_adaptor/pure/v59/research_object_mapping.txt]: As we discussed previous with @fmcc, it provides a default value when the `documents` field is missing from a Pure message. The list projection cannot be applied and results a null value, which then causes downstream problems in the validation. There are some Pure messages which are metadata only and thus do not come with any `documents` arrays.

File: pure_adaptor/pure/base/remapper.py (hunk 4 above)
Comment (camilla-synbyote) [reply pure_adaptor/pure/base/remapper.py]: I think the advantage gained in having all changes in one place is not worth the extra level of indirection that goes along with having to understand the ecs-clusters repo and thread the environment variables through the various terraform files, but I'm not going to argue about this point. I've made the change here and in the ecs clusters repo.

File: pure_adaptor/pure/v59/research_object_mapping.txt (hunk 5 above)
Comment (michalc) [reply pure_adaptor/pure/v59/research_object_mapping.txt]: Ah! I I missed a `}` and thought it was associated with `access`
